Source/WebKit/GPUProcess/graphics/RemoteRenderingBackend.cpp

```diff
@@ -276,8 +276,12 @@ RefPtr<ImageBuffer> RemoteRenderingBackend::allocateImageBuffer(const FloatSize&
     RefPtr<ImageBuffer> imageBuffer;
 
 #if ENABLE(RE_DYNAMIC_CONTENT_SCALING)
-    if (m_gpuConnectionToWebProcess->isDynamicContentScalingEnabled() && (purpose == RenderingPurpose::LayerBacking || purpose == RenderingPurpose::DOM))
+    if (m_gpuConnectionToWebProcess->isDynamicContentScalingEnabled() && (purpose == RenderingPurpose::LayerBacking || purpose == RenderingPurpose::DOM)) {
+        if (!m_dynamicContentScalingResourceCache)
+            m_dynamicContentScalingResourceCache = DynamicContentScalingBifurcatedImageBuffer::createResourceCache();
+        creationContext.dynamicContentScalingResourceCache = m_dynamicContentScalingResourceCache;
```

Comment:
Cannot this be done through an ensure function?
```
creationContext.dynamicContentScalingResourceCache = ensureDynamicContentScalingResourceCache();
```

Reply:
Similar to what is done in `RemoteLayerTreeContext::dynamicContentScalingResourceCache()`.

Reply:
Sure, don't see why not!